src/__tests__/wait-for-dom-change.js

```diff
@@ -0,0 +1,172 @@
+  await new Promise(resolve => setTimeout(resolve, delayMs))
+}
+
+async function skipSomeTimeForMutationObserver(delayMs = 50) {
+  // Using `setTimeout` >30ms instead of `wait` here because `mutationobserver-shim` uses `setTimeout` ~30ms.
+  await skipSomeTime(delayMs, 50)
+}
+
+test('it waits for the next DOM mutation', async () => {
+  const successHandler = jest.fn().mockName('successHandler')
+  const errorHandler = jest.fn().mockName('errorHandler')
+
+  const promise = waitForDomChange().then(successHandler, errorHandler)
+
+  // Promise callbacks are always asynchronous.
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+
+  await skipSomeTimeForMutationObserver()
+
+  // No more expected calls without DOM mutations.
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+
+  document.body.appendChild(document.createElement('div'))
+  expect(document.body).toMatchSnapshot()
+
+  await skipSomeTimeForMutationObserver()
+
+  expect(successHandler).toHaveBeenCalledTimes(1)
+  expect(successHandler).toHaveBeenCalledWith(true)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+  expect(document.body).toMatchSnapshot()
+
+  return promise
+})
+
+test('it waits characterData mutation', async () => {
+  const {container} = render(`<div>initial text</div>`)
+
+  const successHandler = jest.fn().mockName('successHandler')
+  const errorHandler = jest.fn().mockName('errorHandler')
+
+  const promise = waitForDomChange({container}).then(
+    successHandler,
+    errorHandler,
+  )
+
+  // Promise callbacks are always asynchronous.
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+  expect(container).toMatchSnapshot()
+
+  await skipSomeTimeForMutationObserver()
+
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+
+  container.querySelector('div').innerHTML = 'new text'
+  await skipSomeTimeForMutationObserver()
+
+  expect(successHandler).toHaveBeenCalledTimes(1)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+  expect(container).toMatchSnapshot()
+
+  return promise
+})
+
+test('it waits for the attributes mutation', async () => {
+  const {container} = render(``)
+
+  const successHandler = jest.fn().mockName('successHandler')
+  const errorHandler = jest.fn().mockName('errorHandler')
+
+  const promise = waitForDomChange({
+    container,
+  }).then(successHandler, errorHandler)
+
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+
+  await skipSomeTimeForMutationObserver()
+
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+
+  container.setAttribute('data-test-attribute', 'PASSED')
+  await skipSomeTimeForMutationObserver()
+
+  expect(successHandler).toHaveBeenCalledTimes(1)
+  expect(successHandler).toHaveBeenCalledWith(true)
```

Comment:
Why is it called with `true`?

Reply:
Agreed, that's a good question. Why don't we have the success handler be called with the `mutationsList` which the mutation observer will give our callback?

Reply:
This would be a good use case of  toMatchSnapshot right? I was just copying the other ones from waitForElement, but this one would make sense to have I believe.